Lib/idlelib/idle_test/test_config.py

```diff
@@ -26,6 +28,162 @@ def tearDownModule():
+        eq(parser.Get('not', 'exist'), None)
+        eq(parser.Get('not', 'exist', default='DEFAULT'), 'DEFAULT')
+
+    def test_get_option_list(self):
+        parser = config.IdleConfParser('')
+        parser.read_string(self.config)
+
+        self.assertEqual(parser.GetOptionList('one'), ['one', 'two', 'three'])
+        self.assertEqual(parser.GetOptionList('two'), ['one', 'two', 'three'])
+        self.assertEqual(parser.GetOptionList('not exist'), [])
+
+    def test_load_file(self):
+        # Test configfile 'cfgparser.1' borrow from test_configparser
+        config_path = findfile('cfgparser.1')
+        parser = config.IdleConfParser(config_path)
+        parser.Load()
+
+        self.assertEqual(parser.Get('Foo Bar', 'foo'), 'newbar')
+        self.assertEqual(parser.GetOptionList('Foo Bar'), ['foo'])
+
+
+class IdleUserConfParserTest(unittest.TestCase):
+    """Test IdleUserConfParser works"""
+
+    def new_parser(self, path=''):
+        return config.IdleUserConfParser(path)
+
+    def test_add_section(self):
+        parser = self.new_parser()
+        self.assertEqual(parser.sections(), [])
+
+        # Duplicate section should only add one time.
+        # In normal configparser, it will raise DuplicateError,
+        # IdleParser won't raise it
+        parser.AddSection('Foo')
+        parser.AddSection('Foo')
+        parser.AddSection('Bar')
+        s = parser.sections()
+        s.sort()
+        self.assertEqual(s, ['Bar', 'Foo'])
+
+    def test_remove_empty_sections(self):
+        parser = self.new_parser()
+
+        parser.AddSection('Foo')
+        parser.AddSection('Bar')
+        self.assertEqual(sorted(parser.sections()), ['Bar', 'Foo'])
+        parser.RemoveEmptySections()
+        self.assertEqual(parser.sections(), [])
+
+    def test_is_empty(self):
+        parser = self.new_parser()
+
+        parser.AddSection('Foo')
+        parser.AddSection('Bar')
+        self.assertEqual(parser.IsEmpty(), True)
+        self.assertEqual(parser.sections(), [])
+
+        parser.AddSection('Foo')
+        parser.AddSection('Bar')
+        parser.SetOption('Foo', 'bar', 'false')
+        self.assertEqual(parser.IsEmpty(), False)
+        self.assertEqual(parser.sections(), ['Foo'])
+
+    def test_set_options(self):
+        parser = self.new_parser()
+
+        parser.AddSection('Foo')
+
+        # Set option success should return True
+        self.assertEqual(parser.SetOption('Foo', 'bar', 'true'), True)
+
+        # Set duplicate option (with same value) should return False
+        self.assertEqual(parser.SetOption('Foo', 'bar', 'true'), False)
+
+        # Set option and change value should return True
+        self.assertEqual(parser.SetOption('Foo', 'bar', 'false'), True)
+
+        # Set option to not exist section should create section and return True
+        self.assertEqual(parser.SetOption('Bar', 'bar', 'true'), True)
+        self.assertEqual(sorted(parser.sections()), ['Bar', 'Foo'])
```

Comment:
I think using `assertCountEqual()` is OK, `ConfigParser` were using `assertEqual` to test, we only need to care about the item is correct or not here.